interactions/client/bot.py

```diff
@@ -1539,6 +1541,121 @@ async def _logout(self) -> None:
         await self._websocket.close()
         await self._http._req.close()
 
+    async def wait_for(
+        self,
+        name: str,
+        check: Optional[Callable[..., Union[bool, Awaitable[bool]]]] = None,
+        timeout: Optional[float] = None,
+    ) -> Any:
+        """
+        Waits for an event once, and returns the result.
+
+        Unlike event decorators, this is not persistent, and can be used to only proceed in a command once an event happens.
+
+        :param name: The event to wait for
+        :type name: str
+        :param check: A function or coroutine to call, which should return a truthy value if the data should be returned
+        :type check: Callable
+        :param timeout: How long to wait for the event before raising an error
+        :type timeout: float
+        :return: The value of the dispatched event
+        :rtype: Any
+        """
+        while True:
+            fut = self._websocket._dispatch.add(name=name)
+            try:
+                # asyncio's wait_for
+                res: tuple = await wait_for(fut, timeout=timeout)
+            except TimeoutError:
+                with contextlib.suppress(ValueError):
+                    self._websocket._dispatch.extra_events[name].remove(fut)
+                raise
+
+            if check:
+                checked = check(*res)
+                if isawaitable(checked):
+                    checked = await checked
+                if checked:
+                    break
+                else:
+                    # The check failed, so try again next time
+                    log.info(f"A check failed waiting for the {name} event")
+
+        if res:
+            return res[0] if len(res) == 1 else res
+
+    async def wait_for_component(
```

Comment:
can you also make a wait for modal where you unpack the values received automatically and return as a tuple? would be very neat to have it